server/src/main/scala/io/delta/sharing/server/SharedTableManager.scala

```diff
@@ -138,8 +138,14 @@ class SharedTableManager(serverConfig: ServerConfig) {
     val schemaConfig = getSchema(getShareInternal(share), schema)
     getPage(nextPageToken, Some(share), Some(schema), maxResults, schemaConfig.getTables.size) {
       (start, end) =>
-        schemaConfig.getTables.asScala.map { tableConfig =>
-          Table().withName(tableConfig.getName).withSchema(schema).withShare(share)
+        schemaConfig.getTables.asScala.map {
+          tableConfig =>
+            Table(
+              name = Some(tableConfig.getName),
+              schema = Some(schema),
+              share = Some(share),
+              id = Some(tableConfig.getId)
```

Comment:
Should this be optional?

Reply:
What do you mean?

Reply:
The ID in tableConfig has a default value so it means if a table in the config is not configured with id, it'll have an empty value. When a tableConfig is translated here to a proto, a table without ID will have `Some("")` and will produce an empty string in the API response. This behavior will not break any existing usage right?

Reply:
It may not break current usage, unless a client wants to track shares as they are renamed. The field is optional, but it doesn't mean that it should be defaulted to "". The share ID is no longer unique across shares with this.

Reply:
Makes sense. I'll send out a fix. Btw this PR does not add Share ID but only Table ID. I plan to remove the requirement on Share ID on Databricks client.